src/rgw/motr/gc/gc.cc

```diff
@@ -45,23 +45,48 @@ void *MotrGC::GCWorker::entry() {
       iname = motr_gc->index_names[my_index];
       ldpp_dout(dpp, 10) << __func__ << ": " << gc_thread_prefix
         << worker_id << " Working on GC Queue: " << iname << dendl;
-      // time based while loop
-      do {
-        // fetch the next entry from index "iname"
-        // check if the object is ready for deletion
-        // for the motr object
-
-        // delete that object
-        // rc = dequeue();
-
-        // remove the entry from iname
+      
+      bufferlist bl;
+      std::vector<std::string> keys(motr_gc->max_count + 1);
+      std::vector<bufferlist> vals(motr_gc->max_count + 1);
+      uint32_t rgw_gc_obj_min_wait = cct->_conf->rgw_gc_obj_min_wait;
+      rc = motr_gc->store->next_query_by_name(iname,
+                                  keys, vals, obj_tag_prefix);
+      ldpp_dout(dpp, 0) <<__func__<<": next_query_by_name() rc=" << rc << dendl;
+      if (rc < 0) {
+        // In case of failure, worker will keep retrying till end_time
+        ldpp_dout(dpp, 0) <<__func__<<": ERROR: NEXT query failed. rc=" << rc << dendl;
+        continue;
+      }
 
-        // Exit the loop if required work is complete
+      // fetch entries as per defined in rgw_gc_max_trim_chunk from index iname
+      for (uint32_t j = 0; j < motr_gc->max_count && !keys[j].empty(); j++) {
+        bufferlist::const_iterator blitr = vals[j].cbegin();
+        motr_gc_obj_info ginfo;
+        ginfo.decode(blitr);
+        // Check if the object is ready for deletion
+        if(ginfo.deletion_time + rgw_gc_obj_min_wait > std::time(nullptr)) {
+          // No more expired object for deletion
+          break;
+        }
+        // delete motr object
+        rc = motr_gc->delete_motr_obj_from_gc(ginfo);
```

Comment:
delete only if it is simple object

Reply:
This is handled from the producer side, as of now we have diverted only simple object deletion to GC. Multipart objects will still be deleted in sync fashion.

Reply:
I agree  ... but eventually both will be handled via this code only  .. hence was thinking of adding the check rt now only

Reply:
Still in future we will need to distinguish based on ginfo.is_multipart, so we can add some boilerplate for now.
```suggestion
        if (ginfo.is_multipart())
          rc = motr_gc->delete_motr_obj_from_gc(ginfo);
        else
          // handle multipart delete.
```
```

Reply:
Added check for multipart.